src/Sparrow/Json/Parsing/ObjectJsonParser.cs

```diff
@@ -260,6 +259,298 @@ public void Dispose()
                 _ctx.ReturnMemory(_currentStateBuffer);
         }
 
+
+        private enum KnownJsonObjectType : byte
+        {
+            // We introduced a dedicated enum to represent each recognized data shape.
+            // This replaced the repeated, nested 'if (current is X) { ... } else if ...' pattern,
+            // making the parse loop both smaller and easier for the JIT to optimize.
+
+            // Primary object flows  
+            DynamicJsonValue, // DynamicJsonValue => goto ValueTuple
+            ValueTuple,
+
+            // Simple numeric / boolean
+            // (all are quite common and return immediately)
+            Int64,                    // (covers long)
+            Int32,                    // (covers int)
+            Int16OrLesser,            // (covers short, byte, sbyte, ushort)
+            UInt32,                   // (covers uint)
+            UInt64,                   // (covers ulong)
+            Boolean,
+
+            IDynamicJson, // IDynamicJson => goto DynamicJsonValue             
+            BlittableJsonReaderObject, // BlittableJsonReaderObject => goto DynamicJsonValue
+
+            // Lazy string forms
+            LazyStringValue,
+            LazyCompressedStringValue,
+            LazyNumberValue,
+
+            // Character-like => goto handle string
+            Char,
+            EnumType,
+            String,
+
+            // Special handle cases for string / number
+            HandleSetStringBufferOptimization,
+            HandleSetStringBufferNumberOptimization,
+
+            Float, // Float => goto handle number
+            Double, // Double => goto handle number
+            Decimal, // Decimal => can become long or string
+
+#if NET6_0_OR_GREATER
+            Half, // Half => goto handle number
+#endif
+
+            // All datetime based => goto handle string
+            DateTime,
+            DateTimeOffset,
+            TimeSpan,
+#if FEATURE_DATEONLY_TIMEONLY_SUPPORT
+            TimeOnly,
+            DateOnly,
+#endif
+
+            // Collections => goto DynamicJsonArray
+            ListOfLong,
+            ListOfDouble,
+            DictionaryStringLong,
+
+            IEnumerableObject, // IEnumerableObject => goto DynamicJsonArray
+
+            BlittableJsonReaderArray, // BlittableJsonReaderArray => goto DynamicJsonArray
+            DynamicJsonArray, // DynamicJsonArray => base array type
+            BlittableJsonReaderVector, // BlittableJsonReaderVector => goto DynamicJsonArray
+
+            IBlittableJsonContainer, // IBlittableJsonContainer => goto BlittableJsonReaderObject
+            IDynamicJsonValueConvertible,  // Convertible => goto DynamicJsonValue
+            RawBlob,
```

Comment:
Isn't the vector type missing here?

Reply:
I see, yes; but clearly it is either not being called here and/or it is being handled by some of the other ones. I will do the check of adding it and just make it fail on monday to understand exactly what it is happening at this level.